contracts/cw20-reward/src/contract.rs

```diff
@@ -0,0 +1,359 @@
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::ClaimRewards { recipient } => execute_claim_rewards(deps, env, info, recipient),
+        ExecuteMsg::UpdateRewardIndex {} => execute_update_reward_index(deps, env),
+        ExecuteMsg::UnbondStake { amount } => execute_unbound(deps, env, info, amount),
+        ExecuteMsg::WithdrawStake { cap } => execute_withdraw_stake(deps, env, info, cap),
+        ExecuteMsg::Receive(msg) => execute_receive(deps, env, info, msg),
+    }
+}
+
+/// Increase global_index according to claimed rewards amount
+pub fn execute_update_reward_index(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
+    let mut state = STATE.load(deps.storage)?;
+
+    // Zero staking balance check
+    if state.total_balance.is_zero() {
+        return Err(ContractError::NoBond {});
+    }
+
+    // Load the reward contract balance
+    let msg = Cw20QueryMsg::Balance {
+        address: env.contract.address.into_string(),
+    };
+    let query = WasmQuery::Smart {
+        contract_addr: state.cw20_token_addr.clone().into(),
+        msg: to_binary(&msg)?,
+    }
+    .into();
+
+    let balance_res: BalanceResponse = deps.querier.query(&query)?;
+    let previous_balance = state.prev_reward_balance;
+
+    // claimed_rewards = current_balance - prev_balance;
+    let claimed_rewards = balance_res.balance.checked_sub(previous_balance)?;
+
+    state.prev_reward_balance = balance_res.balance;
+
+    // global_index += claimed_rewards / total_balance;
+    state.global_index = state
+        .global_index
+        .add(Decimal::from_ratio(claimed_rewards, state.total_balance));
+
+    STATE.save(deps.storage, &state)?;
+
+    let res = Response::new()
+        .add_attribute("action", "update_reward_index")
+        .add_attribute("claimed_rewards", claimed_rewards)
+        .add_attribute("new_index", state.global_index.to_string());
+
+    Ok(res)
+}
+
+pub fn execute_claim_rewards(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    recipient: Option<String>,
+) -> Result<Response, ContractError> {
+    let recipient = match recipient {
+        Some(value) => deps.api.addr_validate(value.as_str())?,
+        None => info.sender,
+    };
+
+    let mut state = STATE.load(deps.storage)?;
+    let holder = HOLDERS.load(deps.storage, &recipient)?;
+
+    let reward_with_decimals =
+        calculate_decimal_rewards(state.global_index, holder.index, holder.balance)?;
+
+    let all_reward_with_decimals = reward_with_decimals.add(holder.pending_rewards);
+
+    let rewards = all_reward_with_decimals * Uint128::new(1);
+
+    if rewards.is_zero() {
+        return Err(ContractError::NoRewards {});
+    }
+
+    let new_balance = (state.prev_reward_balance.checked_sub(rewards))?;
+    state.prev_reward_balance = new_balance;
+    STATE.save(deps.storage, &state)?;
+
```

Comment:
I think the sending of rewards to the holder is missing, [anchor contract](https://github.com/Anchor-Protocol/anchor-bAsset-contracts/blob/486f1cd495325798f884e19880180a62566b5e48/contracts/anchor_basset_reward/src/user.rs#L58)

Reply:
You are right here! lol what a mistake. I will fix the contract and merge when I have time.